HACKING.md

```diff
@@ -39,3 +39,17 @@ To run the unit tests:
 To run the integration tests:
 
     ./integration-test.sh
+
+### Spark
```

Comment:
This comment doesn't really pertain to your work on this PR, but reading the file over, I feel that there is some general introductory stuff missing. We talk about the various bits of technology we use in this file, but never why or how they fit together. Which up until now was fine, but now that we have spark, a new developer might feel that they need to setup spark in order to setup LB, which isn't the case. Adding more verbiage describing what these bits are used for and why most people won't need to setup spark will make this document much much better.

Reply:
I was assuming that Readme will contain all the information about LB and spark, about how they are related, and how they are not and yes, why it is not necessary to setup both. HACKING.md is more like a file for quick reference to know the correct commands. I am not sure how good of a plan it is. Otherwise HACKING.md can be descriptive. It is just that Readme is first or may be most obvious place to look for detailed setup info. What do you think?

Reply:
Yes, I see that, but now that we've merged the spark portions into this repo, I think we ought to at least put a note into HACKING that lets people know that not everyone needs to setup spark.

And if we want to do a bit of a tech overview in README, that would be fine too -- but right now neither of them really explain how spark fits into LB and that will generate confusion.

Reply:
Sounds good. We can refactor both the files later to make them fit together well. Thanks :)

Reply:
I missed it before but for labs we need to setup LB once. 
https://github.com/metabrainz/listenbrainz-server/blob/master/docker/docker-compose.spark.yml#L47